src/NServiceBus.Core/Persistence/PersistenceStartup.cs

```diff
@@ -17,12 +18,13 @@ public void Run(SettingsHolder settings)
             }
 
             var enabledPersistences = PersistenceStorageMerger.Merge(definitions, settings);
+            var bothFeaturesEnabled = settings.IsFeatureEnabled(typeof(Features.Sagas)) && settings.IsFeatureEnabled(typeof(Features.Outbox));
```

Comment:
the issue with this is that sagas are enabled by default and unless you explicitly disable Sagas, it will always be enabled EVEN when it will be deactivated later during startup because no sagas have been found. This means that this will always be true as soon as you use `config.EnableOutbox()`

Reply:
I think this is good enough. With this additional check the result is that if we do:
```
config.UsePersistence<SomePersistence>();
config.UsePersistence<OtherPersistence, StorageType.Sagas>();
```

it would not throw as we did not enabled the outbox.

If we do

```
config.UsePersistence<SomePersistence>();
config.UsePersistence<OtherPersistence, StorageType.Outbox>();
config.EnableOutbox();
```
it would throw and I think this is OK because potentially you can add a saga by just dropping a DLL to a folder that is scanned. So it is better to tell the users that the storages are incompatible to prevent a nasty surprise later. In that case they should either use the same storage or explicitly ass `DisableFeature<Sagas>` to make is clear that they don't intend to run sagas in this endpoint.

Reply:
good point, you're right with that. It always requires an explicit configuration of either saga or outbox persistence. It would probably make sense to explicitly disable sagas (using `DisableFeature<Sagas>()` in case you want to use the outbox with a different persistence than the one which would be used sagas.

Fine to leave this as is for now 👍